### PR TITLE
[docs]copilot

### PR DESCRIPTION
**Pull request by withshubh**
ref: https://github.com/ToolJet/ToolJet/pull/6074

### CODE REVIEW

**Comment by withshubh** [comment]
**gif pending** @arpitnath

File: docs/docs/copilot.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+<img className="screenshot-full" src="/img/copilot/enable.png" alt="ToolJet Copilot" />
+
+</div>
+
+## Using Copilot
+
+Once activated, the Copilot feature can be accessed while editing any query within the transformations section in the query editor.
+
+In order for Copilot to function, a plain English prompt must be provided within comments. Copilot is capable of manipulating and accessing components, as well as generating code for component specific actions, as it has context of the current state of the application.
```
Comment (arpitnath) [inline docs/docs/copilot.md:28]:
```suggestion
For Copilot to operate, it requires a plain English prompt. Copilot possesses the ability to interact with and access components, along with generating code for specific actions related to those components. This capability is made possible by Copilot's awareness of the application's current state.
```

**Review by arpitnath — CHANGES_REQUESTED** [review]

File: docs/docs/copilot.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
+---
+id: tooljet-copilot
+title: Copilot
+---
+
+:::info BETA
+Tooljet Copilot is currently available in public beta for organizations.
```
Comment (withshubh) [inline docs/docs/copilot.md:7]:
```suggestion
Tooljet Copilot is currently available in public beta for all the users.
```

File: docs/versioned_docs/version-2.5.0/copilot.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+:::info BETA
+Tooljet Copilot is currently available in public beta for organizations.
+:::
+
+**ToolJet Copilot** helps you write your queries faster. It uses OpenAI to suggest queries based on your data. 
+
+## Activating Copilot
+
+To activate Copilot for a ToolJet workspace, the administrator must go to the **workspace settings**, then the **Copilot** section, and toggle on the "**Enable Copilot**" option. Once Copilot is enabled, an **API key** must be provided in order to activate the feature.
```
Comment (arpitnath) [inline docs/versioned_docs/version-2.5.0/copilot.md:14]:
```suggestion
To activate Copilot for a ToolJet workspace, it is necessary to acquire an API key. Once the API key is obtained, Copilot can be enabled by the administrator via the workspace settings. Additionally, users with edit permissions for apps within the workspace can also utilize Copilot.
```

File: docs/versioned_docs/version-2.5.0/copilot.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+You can join the waitlist here: **https://tooljet.com/copilot**
+
+:::info
+- A **new API key** will be needed for each workspace, with *only one user (admin)* allowed per workspace.
```
Comment (arpitnath) [inline docs/versioned_docs/version-2.5.0/copilot.md:19]: can remove this

**Review by arpitnath — CHANGES_REQUESTED** [review]

File: docs/docs/copilot.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
+---
+id: tooljet-copilot
+title: Copilot
+---
+
+:::info BETA
+Tooljet Copilot is currently available in public beta for all the users.
```
Comment (arpitnath) [inline docs/docs/copilot.md:7]:
```suggestion
ToolJet Copilot is currently available in public beta for all the users.
```

File: docs/docs/copilot.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+**ToolJet Copilot** helps you write your queries faster. It uses OpenAI to suggest queries based on your data. 
+
+## Activating Copilot
+
+To access the Copilot, each user, including administrators, can sign up for the waitlist program individually. After signing up, each user will receive a unique API key associated with their account. **Only administrators**, however, will have the privilege to enable Copilot within the workspace settings. Administrators can navigate to the Copilot section in the workspace settings and toggle on the "Enable Copilot" option using their respective API key.
+
+You can join the waitlist here: **https://tooljet.com/copilot**
+
+:::info
+- Copilot can be used by users who have **permissions to edit** the app with a copilot setup.
```
Comment (arpitnath) [inline docs/docs/copilot.md:19]:
```suggestion
- Copilot can be used by users who have **permissions to edit** the app with the copilot setup.
```

**Review by arpitnath — DISMISSED** [review]

File: docs/docs/copilot.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+:::info BETA
+ToolJet Copilot is currently available in public beta for all the users.
+:::
+
+**ToolJet Copilot** helps you write your queries faster. It uses OpenAI to suggest queries based on your data. 
+
+## Activating Copilot
+
+To access the Copilot, each user, including administrators, can sign up for the waitlist program individually. After signing up, each user will receive a unique API key associated with their account. **Only administrators**, however, will have the privilege to enable Copilot within the workspace settings. Administrators can navigate to the Copilot section in the workspace settings and toggle on the "Enable Copilot" option using their respective API key.
```
Comment (tejakummarikuntla) [inline docs/docs/copilot.md:14]: To gain access to Copilot, all users, including administrators, can individually sign up for the waitlist program. Upon successful sign up, each user will be issued a distinctive API key linked to their account. However, the exclusive authority to activate Copilot within the workspace settings lies solely with administrators. Administrators can proceed to the Copilot section in the workspace settings, where they can utilize their respective API key to set the "Enable Copilot" toggle option.

File: docs/docs/copilot.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+<img className="screenshot-full" src="/img/copilot/enable.png" alt="ToolJet Copilot" />
+
+</div>
+
+## Using Copilot
+
+Once activated, the Copilot feature can be accessed while editing any query within the transformations section in the query editor.
+
+For Copilot to operate, it requires a plain English prompt. Copilot possesses the ability to interact with and access components, along with generating code for specific actions related to those components. This capability is made possible by Copilot's awareness of the application's current state.
```
Comment (tejakummarikuntla) [inline docs/docs/copilot.md:32]: Copilot's functionality relies on the provision of clear and concise plain English prompts. It possesses the capability to interact with and retrieve information from various components, enabling it to generate code specific to the desired actions associated with those components. This capability is facilitated by Copilot's comprehensive understanding of the application's present state.

File: docs/docs/copilot.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+### Example: Generating a SQL statement 
+
+We will demonstrate the usage of copilot by generating a SQL query to fetch the order details from a PostgreSQL database based on the key provided in the REST API data.
+
+- Let's create a RESTAPI query using this mock api endpoint: https://fakestoreapi.com/users/1
+- Enable transformation on the query editor
+
```
Comment (tejakummarikuntla) [inline docs/docs/copilot.md:40]: Let's try to avoid statements starting with "Let us", "We will" in the documentation, please stick to a constant voice.

**Review by tejakummarikuntla — COMMENTED** [review]
Added some reviews to improve the delivery of the phrases, to keep it professional try to stick to a constant voice.

-- Teja, Grammarly, ChatGPT

File: docs/docs/copilot.md (hunk 7 above)
Comment (withshubh) [inline docs/docs/copilot.md:14]:
```suggestion
To gain access to Copilot, all users, including administrators, can individually sign up for the waitlist program. Upon successful sign up, each user will be issued a distinctive API key linked to their account. However, the exclusive authority to activate Copilot within the workspace settings lies solely with administrators. Administrators can proceed to the Copilot section in the workspace settings, where they can utilize their respective API key to set the "Enable Copilot" toggle option.
```

File: docs/versioned_docs/version-2.5.0/copilot.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+
+:::info BETA
+Tooljet Copilot is currently available in public beta for all the users.
+:::
+
+**ToolJet Copilot** helps you write your queries faster. It uses OpenAI to suggest queries based on your data. 
+
+## Activating Copilot
+
+To access the Copilot, each user, including administrators, can sign up for the waitlist program individually. After signing up, each user will receive a unique API key associated with their account. **Only administrators**, however, will have the privilege to enable Copilot within the workspace settings. Administrators can navigate to the Copilot section in the workspace settings and toggle on the "Enable Copilot" option using their respective API key.
```
Comment (withshubh) [inline docs/versioned_docs/version-2.5.0/copilot.md:14]:
```suggestion
To gain access to Copilot, all users, including administrators, can individually sign up for the waitlist program. Upon successful sign up, each user will be issued a distinctive API key linked to their account. However, the exclusive authority to activate Copilot within the workspace settings lies solely with administrators. Administrators can proceed to the Copilot section in the workspace settings, where they can utilize their respective API key to set the "Enable Copilot" toggle option.
```

File: docs/docs/copilot.md [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+<img className="screenshot-full" src="/img/copilot/enable.png" alt="ToolJet Copilot" />
+
+</div>
+
+## Using Copilot
+
+Once activated, the Copilot feature can be accessed while editing any query within the transformations section in the query editor.
+
+For Copilot to operate, it requires a plain English prompt. Copilot possesses the ability to interact with and access components, along with generating code for specific actions related to those components. This capability is made possible by Copilot's awareness of the application's current state.
```
Comment (withshubh) [inline docs/docs/copilot.md:32]:
```suggestion
Copilot's functionality relies on the provision of clear and concise plain English prompts. It possesses the capability to interact with and retrieve information from various components, enabling it to generate code specific to the desired actions associated with those components. This capability is facilitated by Copilot's comprehensive understanding of the application's present state.
```

File: docs/versioned_docs/version-2.5.0/copilot.md [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+
+<img className="screenshot-full" src="/img/copilot/enable.png" alt="ToolJet Copilot" />
+
+</div>
+
+## Using Copilot
+
+Once activated, the Copilot feature can be accessed while editing any query within the transformations section in the query editor.
+
+For Copilot to operate, it requires a plain English prompt. Copilot possesses the ability to interact with and access components, along with generating code for specific actions related to those components. This capability is made possible by Copilot's awareness of the application's current state.
```
Comment (withshubh) [inline docs/versioned_docs/version-2.5.0/copilot.md:32]:
```suggestion
Copilot's functionality relies on the provision of clear and concise plain English prompts. It possesses the capability to interact with and retrieve information from various components, enabling it to generate code specific to the desired actions associated with those components. This capability is facilitated by Copilot's comprehensive understanding of the application's present state.
```

File: docs/docs/copilot.md [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+
+## Using Copilot
+
+Once activated, the Copilot feature can be accessed while editing any query within the transformations section in the query editor.
+
+Copilot's functionality relies on the provision of clear and concise plain English prompts. It possesses the capability to interact with and retrieve information from various components, enabling it to generate code specific to the desired actions associated with those components. This capability is facilitated by Copilot's comprehensive understanding of the application's present state.
+
+### Example: Generating a SQL statement 
+
+We will demonstrate the usage of copilot by generating a SQL query to fetch the order details from a PostgreSQL database based on the key provided in the REST API data.
```
Comment (withshubh) [inline docs/docs/copilot.md:36]:
```suggestion
This example demonstrates the usage of copilot by generating a SQL query to fetch the order details from a PostgreSQL database based on the key provided in the REST API data.
```

File: docs/docs/copilot.md [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+
+Once activated, the Copilot feature can be accessed while editing any query within the transformations section in the query editor.
+
+Copilot's functionality relies on the provision of clear and concise plain English prompts. It possesses the capability to interact with and retrieve information from various components, enabling it to generate code specific to the desired actions associated with those components. This capability is facilitated by Copilot's comprehensive understanding of the application's present state.
+
+### Example: Generating a SQL statement 
+
+This example demonstrates the usage of copilot by generating a SQL query to fetch the order details from a PostgreSQL database based on the key provided in the REST API data.
+
+- Let's create a RESTAPI query using this mock api endpoint: https://fakestoreapi.com/users/1
```
Comment (withshubh) [inline docs/docs/copilot.md:38]:
```suggestion
- Create a RESTAPI query using this mock api endpoint: https://fakestoreapi.com/users/1
```

File: docs/versioned_docs/version-2.5.0/copilot.md [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+
+## Using Copilot
+
+Once activated, the Copilot feature can be accessed while editing any query within the transformations section in the query editor.
+
+Copilot's functionality relies on the provision of clear and concise plain English prompts. It possesses the capability to interact with and retrieve information from various components, enabling it to generate code specific to the desired actions associated with those components. This capability is facilitated by Copilot's comprehensive understanding of the application's present state.
+
+### Example: Generating a SQL statement 
+
+We will demonstrate the usage of copilot by generating a SQL query to fetch the order details from a PostgreSQL database based on the key provided in the REST API data.
```
Comment (withshubh) [inline docs/versioned_docs/version-2.5.0/copilot.md:36]:
```suggestion
This example demonstrates the usage of copilot by generating a SQL query to fetch the order details from a PostgreSQL database based on the key provided in the REST API data.
```

File: docs/versioned_docs/version-2.5.0/copilot.md [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+
+Once activated, the Copilot feature can be accessed while editing any query within the transformations section in the query editor.
+
+Copilot's functionality relies on the provision of clear and concise plain English prompts. It possesses the capability to interact with and retrieve information from various components, enabling it to generate code specific to the desired actions associated with those components. This capability is facilitated by Copilot's comprehensive understanding of the application's present state.
+
+### Example: Generating a SQL statement 
+
+This example demonstrates the usage of copilot by generating a SQL query to fetch the order details from a PostgreSQL database based on the key provided in the REST API data.
+
+- Let's create a RESTAPI query using this mock api endpoint: https://fakestoreapi.com/users/1
```
Comment (withshubh) [inline docs/versioned_docs/version-2.5.0/copilot.md:38]:
```suggestion
- Create a RESTAPI query using this mock api endpoint: https://fakestoreapi.com/users/1
```

**Review by tejakummarikuntla — APPROVED** [review]

**Pull request merged**